src/FileConverter.ts

```diff
@@ -31,6 +31,7 @@ export class FileConverter {
             const dialogOptions = { canSelectFiles: true, canSelectMany: false, filters: {} } as OpenDialogOptions;
             const toolExt = FileConverter.Tools.get(tool);
             dialogOptions.filters[`${tool} log files`] = toolExt;
+            dialogOptions.filters['All files'] = [ '*' ]
```

Comment:
>*' [](start = 52, length = 2)

Why is this a good idea? Can't believe I didn't mention this earlier. Tools don't produce output with arbitrary extensions. If you provide a filter for all files, there's no reason to show files of any particular extension (as they will already show). This eliminates that value of, say, reducing a directory with hundreds of files to the subset that are actually relevant to, e.g, a microfocus conversion. What I would have expected is that we added the relevant additional extension for Pylint over showing all files. How does showing a docx file assist a user for any conversion scenario?

Reply:
>  If you provide a filter for all files, there's no reason to show files of any particular extension (as they will already show).

That's not true. If shows the files for the first filter in the list. It only shows "All files" if you select it from the filter drop-down. Do you still object to the "*"?

---
In reply to: [325719497](https://github.com/microsoft/sarif-vscode-extension/pull/201#discussion_r325719497) [](ancestors = 325719497)

Reply:
In addition to the "*", I can add ".plist" to the Fortify extension list, and (for that matter ".sarif.json" to the SARIF extension list. That addresses the customer's spot problem but I still think "*" is good to have for customers who for whatever reason have files with different extensions.

---
In reply to: [325887958](https://github.com/microsoft/sarif-vscode-extension/pull/201#discussion_r325887958) [](ancestors = 325887958,325719497)

Reply:
Yes, let's do what you describe and close this PR. :)

---
In reply to: [325893269](https://github.com/microsoft/sarif-vscode-extension/pull/201#discussion_r325893269) [](ancestors = 325893269,325887958,325719497)

Reply:
@lgolding: You want `sarif.json` over plain `json`?

Reply:
That's a good point. Let's make it plain `.json`. That will pick up any `.sarif.json` files, and I don't see value in excluding `.json` files that _aren't_ `.sarif.json`.

---
In reply to: [406450470](https://github.com/microsoft/sarif-vscode-extension/pull/201#discussion_r406450470) [](ancestors = 406450470)

Reply:
Discussed with Larry offline that `*.json` actually does not apply in the scenario. Thus omitting this aspect of the suggestion.